cmd/launcher/secure_enclave_darwin.go

```diff
@@ -0,0 +1,180 @@
+//go:build darwin
+// +build darwin
+
+package main
+
+import (
+	"crypto"
+	"crypto/ecdsa"
+	"crypto/rand"
+	"encoding/base64"
+	"errors"
+	"fmt"
+	"os"
+	"time"
+
+	"github.com/kolide/krypto/pkg/challenge"
+	"github.com/kolide/krypto/pkg/echelper"
+	"github.com/kolide/krypto/pkg/secureenclave"
+	"github.com/kolide/launcher/ee/agent/certs"
+	"github.com/kolide/launcher/ee/secureenclavesigner"
+	"github.com/vmihailenco/msgpack/v5"
+)
+
+const secureEnclaveTimestampValiditySeconds = 150
+
+var serverPubKeys = make(map[string]*ecdsa.PublicKey)
+
+// runSecureEnclave performs either a create-key or sign operation using the secure enclave.
+// It's available as a separate command because launcher runs aa root by default and since it's
+// not in a user security context, it can't use the secure enclave directly. However, this command
+// can be run in the user context using launchctl.
```

Comment:
The documentation here definitely helps -- I think it would be good to include an example of what root launcher would need to use this for, too (e.g. what kind of data it will sign)